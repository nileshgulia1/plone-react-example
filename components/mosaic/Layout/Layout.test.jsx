import React from 'react';
import renderer from 'react-test-renderer';
import Layout from './Layout';

jest.mock('../Grid/Grid', () => jest.fn(() => <div />));
jest.mock('../Editbar/Editbar', () => jest.fn(() => <div />));

global.__SERVER__ = true; // eslint-disable-line no-underscore-dangle

describe('Layout', () => {
  it('renders a layout component', () => {
    const component = renderer.create(
      <Layout
        layout={{
          rows: [
            {
              columns: [
                {
                  width: 16,
                  tiles: [
                    {
                      content: '<p>Column <b>one</b></p>',
                      url: './@@plone.app.standardtiles.html/1',
                      type: 'title',
                    },
                  ],
                },
              ],
            },
          ],
        }}
      />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
