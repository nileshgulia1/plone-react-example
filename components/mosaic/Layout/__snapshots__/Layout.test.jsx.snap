// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders a layout component 1`] = `
<div>
  <div />
  <div />
</div>
`;
